data_scrapers/sonoma.py

```diff
@@ -0,0 +1,299 @@
+    categories = []
+    rows = get_rows(tag)
+    for row in rows:
+        group, cases, _pct = get_cells(row)
+        raw_cases = parse_int(cases)
+        age_string_transform = {
+            '0-17': '0_to_17',
+            '18-49': '18_to_49',
+            '50-64': '50_to_64',
+            '65 and Above': '65_and_older',
+            'Under Investigation': 'Unknown'
+        }
+
+        if group not in age_string_transform:
+            raise FormatError('A new age group has been added to the cases by race table')
+
+        element = {'group': age_string_transform[group], 'raw_cases': raw_cases}
+        categories.append(element)
+    return categories
+
+def get_unknown_race(race_eth_tag: element.Tag) -> int:
+    """
+    Gets the notes under the 'Cases by race and ethnicity' table to find the
+    number of cases where the person's race is unknown
+    """
+    parent = race_eth_tag.parent
+    note = parent.find('p').text
+    matches = re.search('(\d+) \(\d{1,3}%\) missing race/ethnicity', note)
+    if not matches:
+        raise FormatError('The format of the note with unknown race data has changed')
+    return(parse_int(matches.groups()[0]))
+
+def transform_race_eth(race_eth_tag: element.Tag) -> Dict[str, int]:
+    """
+    Takes in the BeautifulSoup tag for the cases by race/ethnicity table and
+    transforms it into an object of form:
+    'race_eth': {'Asian': -1, 'Latinx_or_Hispanic': -1, 'Other': -1, 'White':-1, 'Unknown': -1}
+    NB: These are the only races reported seperatley by Sonoma county at this time
+    """
+    race_cases = {
+        'African_Amer': 0,
+        'Asian': 0,
+        'Latinx_or_Hispanic': 0,
+        'Native_Amer':0,
+        'Multiple_Race':0,
+        'Other': 0,
+        'Pacific_Islander': 0,
+        'White': 0,
+        'Unknown': 0
+    }
+    race_transform = {'Asian/Pacific Islander, non-Hispanic': 'Asian', 'Hispanic/Latino': 'Latinx_or_Hispanic', 'Other*, non-Hispanic': 'Other', 'White, non-Hispanic': 'White'}
+    rows = get_rows(race_eth_tag)
+    for row in rows:
+        group_name, cases, _pct = get_cells(row)
+        if group_name not in race_transform:
+            raise FormatError('The racial group {0} is new in the data -- please adjust the scraper accordingly')
+        internal_name = race_transform[group_name]
+        race_cases[internal_name] = parse_int(cases)
+    race_cases['Unknown'] = get_unknown_race(race_eth_tag)
+    return race_cases
+
+def transform_total_hospitalizations(hospital_tag: element.Tag) -> Dict[str, int]:
+    """
+    Takes in a BeautifulSoup tag of the cases by hospitalization table and
+    returns a dictionary with the numbers of hospitalized and non-hospitalized
+    cases
+    """
+    hospitalizations = {}
+    rows = get_rows(hospital_tag)
+    for row in rows:
+        hospitalized, number, _pct = get_cells(row)
+        if hospitalized.lower() == 'yes':
+            hospitalizations['hospitalized'] = parse_int(number)
+        else:
```

Comment:
Might be worth checking the value here anyway. (And also raising an error is `hospitalized.lower()` is something totally unexpected.) What if the row titles change to `Hospitalized` and `Not Hospitalized`? (Or something else.) Then we’d wind up with:

```js
{
  // Note "hospitalized" is not even present.
  "not_hospitalized": <value of last row in table, regardless of what it represents>
}
```